### PR TITLE
[Inductor] Skip triton templates for mixedmm on SM70-

### LINKED ISSUE #117144: `test_mixed_mm2_cuda` fails on V100
### 🐛 Describe the bug

Tested in a source build and the latest nightly+cu121.
Steps to reproduce the issue on a V100:
```python
python inductor/test_torchinductor.py -v -k test_mixed_mm2_cuda
test_mixed_mm2_cuda (__main__.CudaTests) ... FAIL

======================================================================
FAIL: test_mixed_mm2_cuda (__main__.CudaTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/torch/testing/_internal/common_utils.py", line 2674, in wrapper
    method(*args, **kwargs)
  File "/workspace/src/pytorch/test/inductor/test_torchinductor.py", line 8197, in new_test
    return value(self)
  File "/usr/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/workspace/src/pytorch/test/inductor/test_torchinductor.py", line 2136, in test_mixed_mm2
    self.common(
  File "/usr/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/workspace/src/pytorch/test/inductor/test_torchinductor.py", line 516, in check_model_cuda
    check_model(
  File "/workspace/src/pytorch/test/inductor/test_torchinductor.py", line 376, in check_model
    self.assertEqual(
  File "/usr/local/lib/python3.10/dist-packages/torch/testing/_internal/common_utils.py", line 3518, in assertEqual
    raise error_metas.pop()[0].to_error(
AssertionError: Tensor-likes are not close!

Mismatched elements: 32 / 64 (50.0%)
Greatest absolute difference: 556.0 at index (4, 6) (up to 1e-05 allowed)
Greatest relative difference: 2.876953125 at index (2, 7) (up to 0.001 allowed)
```
I'm not sure if these Trriton kernels ever worked on Volta as it seems the tests were never executed before: https://github.com/pytorch/pytorch/pull/114504

Attaching the created code from `TORCH_LOGS=output_code python inductor/test_torchinductor.py -v -k test_mixed_mm2_cuda` assuming it makes it easier to isolate the issue.
[mixed_mm2_code.txt](https://github.com/pytorch/pytorch/files/13893887/mixed_mm2_code.txt)

### Versions

`2.3.0.dev20240110+cu121`

CC @eqy 

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @muchulee8 @aakhundov @ColinPeppler

### PR DESCRIPTION
**Pull request by malfet**
As it results in numerical errors, see https://github.com/pytorch/pytorch/issues/117144

Fixes https://github.com/pytorch/pytorch/issues/117144

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by malfet** [comment]
I wonder if I should move `is_v100_gpu` to utils (and perhaps rename it to `is_volta_gpu`?

File: torch/_inductor/kernel/mm.py [hunk 1]
```diff
@@ -259,11 +260,19 @@ def fallback_mixed_mm(mat1, mat2, *, out):
 aten_fallback_mixed_mm = ExternKernelChoice(fallback_mixed_mm, None)
 
 
+@functools.lru_cache(None)
+def _is_v100_gpu(index: Optional[int]) -> bool:
+    props = torch.cuda.get_device_properties(index or 0)
+    return props.major == 7
```
Comment (jansel) [inline torch/_inductor/kernel/mm.py:266]:
```suggestion
    return props.major <= 7
```

File: torch/_inductor/kernel/mm.py [hunk 2]
```diff
@@ -259,11 +260,19 @@ def fallback_mixed_mm(mat1, mat2, *, out):
 aten_fallback_mixed_mm = ExternKernelChoice(fallback_mixed_mm, None)
 
 
+@functools.lru_cache(None)
+def _is_v100_gpu(index: Optional[int]) -> bool:
```
Comment (jansel) [inline torch/_inductor/kernel/mm.py:264]:
```suggestion
def _is_v100_or_older_gpu(index: Optional[int]) -> bool:
```

**Review by jansel — CHANGES_REQUESTED** [review]

**Comment by jansel** [comment]
Seems safer to also disable for older GPUs than Volta.

**Comment by malfet** [comment]
@jansel renamed to `_is_sm7x_or_older_gpu` (I wonder if I should just return False for ROCM GPUs)

**Review by jansel — APPROVED** [review]

**Comment by jansel** [comment]
Not sure what to do about rocm.  Maybe someone from AMD can help figure out minimum requirements there.

**Pull request closed without merging**